src/Plugin/Validation/Constraint/GalleryMediaBundleConstraintValidator.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Drupal\media\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the GalleryMediaBundle constraint.
+ */
+class GalleryMediaBundleConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+    if (!isset($value)) {
+      return;
+    }
+    if ($value->hasField($constraint->sourceFieldName)) {
+      $slideshowItems = $value->get($constraint->sourceFieldName);
+      foreach ($slideshowItems as $item) {
+        $itemBundle = \Drupal::entityTypeManager()->getStorage('media')->load($item->target_id)->bundle();
+        if ($itemBundle == "gallery") {
```

Comment:
I wouldn't hardcode bundle machine name. Instead of that I'd check the type plugin that is used with it.